src/app/app.component.js

```diff
@@ -30,15 +31,35 @@ function AppComponentCtrl(
 
     this.settings = dimSettingsService;
     $scope.$watch(() => this.settings.itemSize, (size) => {
-      document.querySelector('html').style.setProperty("--item-size", `${size}px`);
+      if (!isPhonePortrait()) {
+        document.querySelector('html').style.setProperty("--item-size", `${size}px`);
+      }
     });
     $scope.$watch(() => this.settings.charCol, (cols) => {
-      document.querySelector('html').style.setProperty("--character-columns", cols);
+      if (!isPhonePortrait()) {
+        document.querySelector('html').style.setProperty("--character-columns", cols);
+      }
     });
     $scope.$watch(() => this.settings.vaultMaxCol, (cols) => {
       document.querySelector('html').style.setProperty("--vault-max-columns", cols);
     });
 
+    $scope.$watch(() => this.settings.charColMobile, (cols) => {
+      if (isPhonePortrait()) {
+        document.querySelector('html').style.setProperty("--character-columns", cols);
+      }
+    });
+
+    // a watch on isPhonePortrait is needed when the user on mobile changes from portrait to landscape
+    // or a user on desktop shrinks the browser window below isphoneportrait treshold value
+    $scope.$watch(() => isPhonePortrait(), (isPhonePortrait) => {
```

Comment:
This isn't the right way to do this - it'll evaluate the media rule on every digest! Instead, subscribe to the Observable of isPhonePortraitStream()